### ISSUE TITLE
Memory exhausted while uploading a large torrent

### ISSUE THREAD

**Issue by bianyifan** [Rhilip/RidPT#1]
![image](https://user-images.githubusercontent.com/12882489/52160546-8497e900-26f2-11e9-916a-efdfee4b461d.png)

The torrent file uploaded is 1.2 MiB large. Because GitHub does not support attachments of issues, the related torrent file has been sent to the author in another way.

**Comment by povsister** [comment]
In fact, NexusPHP has every uploaded torrent file verified. It uses a set of recursive functions to encode/decode bencoded files.
One ill-formatted element encountered during the parse is enough to cause a NULL return val, which means something went wrong. And the server just simply reject such torrents with those erros.

It looks not that elegant but really works out on most occasion.

**Comment by bianyifan** [comment]
That specific torrent file is valid except trailing new line. That new line (0x0a, aka CR) is automatically added by vim on saving file after I changed the announce URL of that torrent.

Still working on memory exhausting issue.

**Comment by Rhilip** [comment]
It does not seem to reappear after commit https://github.com/Rhilip/RidPT/commit/4413b46daad897674c82ce359b9a1d193db83188 , Though I think this issues fix after the release `v0.1.2-alpha`

With Test Setting :
```
'worker_num'       => 1,        // 工作进程数
'max_request'      => 3000,   // 进程的最大任务数
```

After three times trys to upload this large torrent , The Memory used not exhausted , the error report show this value change like `4436288 -> 4625136 -> 4625136`.
Also I notice When increase the setting of `worker_num` to default number `8`, The fail tmp_file will be autocleaned after this worker next time when it recevie the new request.

**Comment by Rhilip** [comment]
I don't think this bug can reproduce after 5 alpha versions, since many change has been commited. 
So I decide to closed it.
